src/Components/Web/src/Forms/InputChoice.cs

```diff
@@ -0,0 +1,46 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Diagnostics.CodeAnalysis;
+using System.Globalization;
+
+namespace Microsoft.AspNetCore.Components.Forms
+{
+    /// <summary>
+    /// Serves as a base for inputs that have a group of selectable choices.
+    /// </summary>
+    public abstract class InputChoice<TValue> : InputBase<TValue>
```

Comment:
Do you think we could start off by not having this type? We could find other ways to share the code, for instance having an non-public helper type.